mithril-aggregator/src/multi_signer.rs

```diff
@@ -703,34 +703,54 @@ impl MultiSigner for MultiSignerImpl {
             .get_protocol_parameters()
             .await?
             .ok_or_else(ProtocolError::UnavailableProtocolParameters)?;
-        let avk = &self
+
+        let clerk = self
             .clerk
             .as_ref()
-            .ok_or_else(ProtocolError::UnavailableClerk)?
-            .compute_avk();
+            .ok_or_else(ProtocolError::UnavailableClerk)?;
 
-        signatures
+        let signature = signatures
             .to_protocol_signature()
-            .map_err(ProtocolError::Codec)?
-            .verify(&protocol_parameters, avk, message.compute_hash().as_bytes())
-            .map_err(|e| ProtocolError::Core(e.to_string()))?;
-
-        // Register single signature
-        let beacon = self
-            .current_beacon
-            .as_ref()
-            .ok_or_else(ProtocolError::UnavailableBeacon)?;
+            .map_err(ProtocolError::Codec)?;
+
+        let avk = clerk.compute_avk();
+
+        // If there is no reg_party, then we simply received a signature from a non-registered
+        // party, and we can ignore the request.
+        if let Some((vk, stake)) = clerk.get_reg_party(&signature.signer_index) {
```

Comment:
```suggestion
        let (vk, stake) = clerk.get_reg_party(&signature.signer_index).ok_or_else(|| ProtocolError::UnregisteredParty)?
```
This should avoid to create the `if let` and would return an error instead of `Ok(())`